frontend/server/src/Controllers/Problem.php

```diff
@@ -4742,6 +4743,30 @@ public static function getProblemListForSmarty(
         ];
     }
 
+    /**
+     *
+     * @return array{smartyProperties: array{payload: ProblemListCollection, title: string}, entrypoint: string}
+     *
+     */
+    public static function getProblemCollectionDetailsForSmarty(
+        \OmegaUp\Request $r
+    ): array {
+        return [
+            'smartyProperties' => [
+                'payload' => [
+                    'levelTags' => \OmegaUp\Controllers\Tag::getLevelTags(),
+                    'problemsAmount' => \OmegaUp\DAO\Problems::getAmountOfProblems(
+                        $r['problemLevelTag']
```

Comment:
Por convención, usamos snake_case en los parámetros que se mandan vía request:
```suggestion
                        $r['problem_level_tag']
```
Pero como `getAmountOfProblems` espera un string y aquí no sabemos que tipo de dato se está mandando, puedes usar:
```
            $r->ensureString(
                'problem_level_tag',
                fn (string $problemLevelTag) => \OmegaUp\Validators::stringNonEmpty(
                    $problemLevelTag
                )
            );
```
para asegurar que el parámetro es una cadena y que no está vacía.

Y para que pueda ser leído desde aquí, tendrías que declararlo en el archivo `nginx.rewrites` algo así:
```
rewrite ^/problem/collection/([a-zA-Z0-9_-]+)/?$ /problems/collections.php?problem_level_tag=$1 last;
```